### PR TITLE
WOR-828 consistent use of region and location

### PR DESCRIPTION
**Pull request by dvoet**
https://broadworkbench.atlassian.net/browse/WOR-828

### CODE REVIEW

File: service/src/test/java/bio/terra/policy/service/region/RegionServiceTest.java [hunk 1]
```diff
@@ -74,33 +74,33 @@ void locationContainsRegionNegative() {
 
   @Test
   void getRegionsForLocationInvalidRegion() {
-    var result = regionService.getRegionsForLocation("invalid", GCP_PLATFORM);
+    var result = regionService.getRegionCodesForLocation("invalid", GCP_PLATFORM);
```
Comment (ddietterich) [inline service/src/test/java/bio/terra/policy/service/region/RegionServiceTest.java:77]: Why add "code"? What is the difference between a region and a region code?

**Review by ddietterich — COMMENTED** [review]
This is tidy. One question about the term "regionCode".
We should document the terminology somewhere.

File: service/src/test/java/bio/terra/policy/service/region/RegionServiceTest.java (hunk 1 above)
Comment (dvoet) [reply service/src/test/java/bio/terra/policy/service/region/RegionServiceTest.java:77]: The Region object has a code which is different than its id. I am trying to be clear what is being returned. Returning the Region itself is probably clearer. All these Strings are confusing anyway.

**Review by mikenev — APPROVED** [review]
This is a lot tidier, thank you.

**Comment by dvoet** [comment]
@ddietterich @mikenev PTAL - I made the RegionService more strongly typed (less Strings). I also leaned into streams a little more.

**Review by ddietterich — APPROVED** [review]
Thanks a lot, Doug!

**Review by mikenev — APPROVED** [review]
thank you!

**Pull request merged**